competitors/fst_wrapper.h

```diff
@@ -53,20 +53,14 @@ class FST : public Competitor {
       key = std::string(reinterpret_cast<const char*>(&endian_swapped_word), 4);
     }
 
-    uint64_t guess;
+    uint64_t guess = 0;
     if (lookup_key >= max_key_) {
       // looking up a value greater than the largest value causes a segfault...
       return (SearchBound){max_val_, data_size_};
       std::cout << max_val_ << "!!!" << std::endl;
     } else {
-      auto iter = fst_->moveToKeyGreaterThan(key, true);
-
-      // sometimes we get back a bad iterator even though
-      // we shouldn't...
-      if (!iter.isValid()) return (SearchBound){0, data_size_};
-
-      // multiply by size_scale here because getValue() returns an index
-      guess = iter.getValue() * size_scale;
+      if (!fst_->lookupKey(key, guess)) return {0, data_size_};
```

Comment:
That doesn't look correct. `lookupKey` only works if we look up keys, not non-keys. The previous code did a lower-bound lookup, which is what we should be doing. Here, we fall back to a full binary search for non-keys, which means we benchmark binary search instead of FST.

Reply:
Oh no, you are of course correct. Thank you for catching and pointing this out!  

I have to admit I did not properly think about this since I copy and pasted code from another repository in which I always do full keyset indexing, i.e., lookupKey() always works. Plus the benchmarks I ran seem to indicate (though only evaluated using _eye ball statistics_) that the new code actually runs faster than the old code - even if it defaults to full binary search in most cases. Whoops!

I will shortly test to verify that at least the full indexing case is indeed faster using lookupKey() and update this PR such that only the full indexing fst uses lookupKey() and everything else defaults to the previous code